redash/query_runner/elasticsearch.py

```diff
@@ -252,4 +285,85 @@ def run_query(self, query):
         return json_data, error
 
 
+#
+#
+# Simple query example:
+#
+# - Query the index named "twitter"
+# - Filter by user equal "kimchy"
+# - Return the fields: "@timestamp", "tweet" and "user"
+# - Return up to 15 results
+# - Sort by @timestamp ascending
+#
+# {
+#     "index" : "twitter",
+#     "query" : {
+#       "match": {
+#           "user" : "kimchy"
+#       }
+#     },
+#     "fields" : ["@timestamp", "tweet", "user"],
+#     "size" : 15,
+#     "sort" : "@timestamp:asc"
+# }
+#
+#
+
+class ElasticSearchV2(BaseElasticSearch):
+
+    def __init__(self, configuration_json):
+        super(ElasticSearchV2, self).__init__(configuration_json)
+
+    @classmethod
+    def enabled(cls):
+        return True
+
+    @classmethod
+    def annotate_query(cls):
+        return False
+
+    def run_query(self, query):
+        try:
+            error = None
+
+            logger.debug(query)
+            query_params = json.loads(query)
+
+            index_name = query_params.pop("index", "")
+
+            if not self.server_url:
+                error = "Missing configuration key 'server'"
+                return None, error
+
+            url = "{0}/{1}/_search".format(self.server_url, index_name)
+            mapping_url = "{0}/{1}/_mapping".format(self.server_url, index_name)
+
+            mappings = self._get_mappings(mapping_url)
+
+            logger.debug(json.dumps(mappings, indent=4))
+
+            logger.debug("Using URL: {0}".format(url))
+            logger.debug("Using Query: {0}".format(json.dumps(query_params)))
```

Comment:
Using `.format` method here means that even if the logging level is higher than debug, the string will be created. Use the old format: `logger.debug("Using Query: %s", query_params)` (I've also removed the `json.dumps` call, because otherwise it will get serialized every time regardless of logging level).